From e90ef1686c6b0a644282ec6da137a186fbc3ddb7 Mon Sep 17 00:00:00 2001
From: Christophe Chapuis <chris.chapuis@gmail.com>
Date: Tue, 25 Jul 2017 18:01:28 +0000
Subject: [PATCH 09/10] Disable some sandboxing capabilities

LuneOS is running on old kernels (< 3.8), so disable some sandboxing
functionalities that can be problematic.

Signed-off-by: Christophe Chapuis <chris.chapuis@gmail.com>
---
 chromium/content/common/sandbox_linux/sandbox_seccomp_bpf_linux.cc | 3 +++
 chromium/sandbox/linux/services/credentials.cc                     | 5 +++++
 2 files changed, 8 insertions(+)

diff --git a/chromium/content/common/sandbox_linux/sandbox_seccomp_bpf_linux.cc b/chromium/content/common/sandbox_linux/sandbox_seccomp_bpf_linux.cc
index e8264cf..aa12ccd 100644
--- a/chromium/content/common/sandbox_linux/sandbox_seccomp_bpf_linux.cc
+++ b/chromium/content/common/sandbox_linux/sandbox_seccomp_bpf_linux.cc
@@ -218,6 +218,9 @@ bool StartBPFSandbox(const base::CommandLine& command_line,
 
 // Is seccomp BPF globally enabled?
 bool SandboxSeccompBPF::IsSeccompBPFDesired() {
+  // LuneOS: SECCOMP_FILTER is not available for ARM on 3.4 kernels
+  return false;
+    
   const base::CommandLine& command_line =
       *base::CommandLine::ForCurrentProcess();
   if (!command_line.HasSwitch(switches::kNoSandbox) &&
diff --git a/chromium/sandbox/linux/services/credentials.cc b/chromium/sandbox/linux/services/credentials.cc
index 2265474..01143ed 100644
--- a/chromium/sandbox/linux/services/credentials.cc
+++ b/chromium/sandbox/linux/services/credentials.cc
@@ -253,6 +253,11 @@ bool Credentials::HasCapability(Capability cap) {
 
 // static
 bool Credentials::CanCreateProcessInNewUserNS() {
+  // LuneOS hack: as detection of user_namespace capability is incorrect
+  // (doesn't match the actual code in 
+  // NamespaceUtils::KernelSupportsUnprivilegedNamespace), disable it.
+  return false;
+    
   // Valgrind will let clone(2) pass-through, but doesn't support unshare(),
   // so always consider UserNS unsupported there.
   if (IsRunningOnValgrind()) {
-- 
2.7.4

